src/aleph_vrf/coordinator/executor_selection.py

```diff
@@ -119,6 +119,21 @@ async def select_executors(self, nb_executors: int) -> List[Executor]:
             raise NotEnoughExecutors(requested=nb_executors, available=len(executors))
         return random.sample(executors, nb_executors)
 
+    async def get_candidate_executors(self) -> List[Executor]:
+        compute_nodes = self._list_compute_nodes()
+        blacklisted_nodes = self._get_unauthorized_nodes()
+        whitelisted_nodes = (
+            node
+            async for node in compute_nodes
+            if node.address not in blacklisted_nodes
+        )
+        executors = [
+            AlephExecutor(node=node, vm_function=self.vm_function)
+            async for node in whitelisted_nodes
+        ]
```

Comment:
Why not combine the two logics as one ?

```suggestion
        executors = [
            AlephExecutor(node=node, vm_function=self.vm_function)
            async for node in compute_nodes
            if node.address not in blacklisted_nodes
        ]
```

Reply:
Just to have it separetely, allow to block nodes, and allow just to whitelist nodes.

Reply:
The `whitelisted_nodes` variable is not used anywhere else at the moment, so the two versions achieve the exact same result. Can you be more clear on the added value of the additional iteration ?

Reply:
Okay, I miss-understood the question. This code was done by Olivier 5 months ago and I just copied it. Also I guessed that was another part of the code. Now is fixed in both places to follow the same logic.